enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,321 @@
+---
+title: csi-driver-operator-merge
+authors:
+  - jsafrane
+reviewers:
+  - "@storage-team"
+  # TODO: someone from ART
+  # TODO: staff engineer for new image names
+approvers:
+  - TBD
+api-approvers:
+  - None
+creation-date: 2023-09-05
+last-updated: 2023-09-05
+tracking-link:
+  - https://issues.redhat.com/browse/STOR-1437
+see-also:
+  - "/enhancements/storage/csi-driver-install.md"
+replaces:
+superseded-by:
+---
+
+# Merge CSI driver operators into a single git repository
+
+## Summary
+
+CSI driver operators are OCP specific operators that install and manage CSI drivers shipped by OCP either as part of
+payload or as optional operators installed by OLM.
+
+In this document we propose that all these operators are compiled from a single git repository, as they're mostly
+copy-paste.
+
+## Motivation
+
+OpenShift comes with these CSI driver operators built by ART and either shipped as part of the OpenShift
+payload or via OLM:
+
+* [Alibaba Disk](https://github.com/openshift/alibaba-disk-csi-driver-operator/) (*)
+* [AWS EBS](https://github.com/openshift/aws-ebs-csi-driver-operator/)
+* [AWS EFS](https://github.com/openshift/aws-efs-csi-driver-operator/) (OLM)
+* [Azure Disk](https://github.com/openshift/azure-disk-csi-driver-operator/)
+* [Azure File](https://github.com/openshift/azure-file-csi-driver-operator/)
+* [GCP PD](https://github.com/openshift/gcp-pd-csi-driver-operator/)
+* [GCP Filestore](https://github.com/openshift/gcp-filestore-csi-driver-operator/) (OLM)
+* [IBM PowerVS](https://github.com/openshift/ibm-powervs-block-csi-driver-operator/)
+* [IBM VPC](https://github.com/openshift/ibm-vpc-block-csi-driver-operator/)
+* [OpenStack Cinder](https://github.com/openshift/openstack-cinder-csi-driver-operator/)
+* [OpenStack Manila](https://github.com/openshift/csi-driver-manila-operator/)
+* [oVirt Disk](https://github.com/openshift/ovirt-csi-driver-operator/) (*)
+* [Secret Store](https://github.com/openshift/secrets-store-csi-driver-operator/) (OLM)
+* [Shared Resource](https://github.com/openshift/csi-driver-shared-resource-operator/)
+* [VMware vSphere](https://github.com/openshift/vmware-vsphere-csi-driver-operator/)
+
+Operators marked by (*) will be removed soon, as OCP stops supporting the particular cloud provider.
+
+All these operators have their own separate github repository (i.e. 15 of them). All of them share the
+same [CSI driver operator library in library-go](https://github.com/openshift/library-go/tree/master/pkg/operator/csi),
+so their code is quite small and mostly just configure the shared library.
+
+* When a CVE is found in the shared library, it needs to be fixed in all 15 repositories. Similarly, Kubernetes go
+  packages (like k8s.io/client-go) needs to be bumped in each of them. This is a lot of work, and it's easy to miss one
+  of the repositories.
```

Comment:
We should emphasize that this applies to every new feature or bug fix in the library-go, and there are numerous instances of this throughout a release

Reply:
Emphasized :-)